massgov/mayflower
# DP-20555 Add a skip link to figure component

Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/develop/docs/for-developers/changelog-instructions.md)


## Description

- Add a skip link right above the <figure> to give keyboard and screen reader users an option to skip the figure content.
- Set up to process a flag value to determine whether add a skip link or not to the template along with its default value based on content type embedded to the figure container:
          - Tableau:  true
          - Caspio:  false
          - iframe:  false
          - image:  false
- Make a skip link visible only when it's get focus.
- Skip link text has its default value, which can be overridden if another value is available.
- Skip link aria-label has figure content title in addition to the initial link text to better identify the figure component.
- Set up a skip link target text to be visible when a link is clicked to clarify whereabout of users in the page.
- Skip link target text has its default value, which can be overridden if another value is available.
- Skip link target aria-label has figure content title in addition to the initial link text to better identify the figure component.
- Hide the skip link target text as they move on to next/previous element.

## Related Issue / Ticket

- [JIRA issue](https://jira.mass.gov/browse/DP-20555)
- [Github issue](https://github.com/massgov/openmass/pull/570)

## Steps to Test


1.  Go to figure template variations.
2. In inspect, find their skip links and their targets are added.  Note:  For sample presentation purpose, they are set to add their skip links.
3. Go to the data viz samples under the pages.  Find: 

- Find the skip link appears as it gets focus.
- Find the skip link takes you to its paired skip link target when the link is clicked, and the skip link disappears.
- A skip link doesn't shift the page content as it's displayed when it doesn't go over one line.
- When a figure component is aligned to right, its skip link and its link target text are also alined to the left edge of the component.
<img width="795" alt="align" src="https://user-images.githubusercontent.com/9633303/103140368-b5e97b80-46b3-11eb-91aa-803144480bd7.png">
<img width="777" alt="align_target" src="https://user-images.githubusercontent.com/9633303/103140370-ba159900-46b3-11eb-9870-a7cc32969920.png">

- When a skip link is clicked, its target text is aligned below the sticky nav.  It doesn't get covered by the sticky nav.
<img width="699" alt="Pattern_Lab_-_pages-information-details-dataviz-large-1col" src="https://user-images.githubusercontent.com/9633303/103140420-59d32700-46b4-11eb-92cb-a72d160d4d33.png">

- As moving forward/backward from the skip link target, the skip link target disappears.

 

## Screenshots
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.


## Additional Notes:

Anything else to add?

#### Impacted Areas in Application


*

#### @TODO


*

#### Today I learned...
